From bb571f6c8d4f0ae019783c9fc68c9e781aaab0d1 Mon Sep 17 00:00:00 2001
From: Balint Erdi <balint.erdi@gmail.com>
Date: Thu, 16 Apr 2009 11:11:07 +0200
Subject: [PATCH] have '_mailer' appended to directory and -some- file names that were generated with ./script/generate mailer

Also, the mailer class will have the name <class_name>Mailer and the unit test <class_name>MailerTest

Signed-off-by: Balint Erdi <balint.erdi@gmail.com>
---
 .../components/mailer/mailer_generator.rb          |   13 +++++++------
 .../components/mailer/templates/mailer.rb          |    2 +-
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/railties/lib/rails_generator/generators/components/mailer/mailer_generator.rb b/railties/lib/rails_generator/generators/components/mailer/mailer_generator.rb
index ba6d60c..6fdfd0f 100644
--- a/railties/lib/rails_generator/generators/components/mailer/mailer_generator.rb
+++ b/railties/lib/rails_generator/generators/components/mailer/mailer_generator.rb
@@ -2,21 +2,22 @@ class MailerGenerator < Rails::Generator::NamedBase
   def manifest
     record do |m|
       # Check for class naming collisions.
-      m.class_collisions class_name, "#{class_name}Test"
+      m.class_collisions class_name, "#{class_name}MailerTest"
 
+      file_path_with_mailer = file_path + '_mailer'
       # Mailer, view, test, and fixture directories.
       m.directory File.join('app/models', class_path)
-      m.directory File.join('app/views', file_path)
+      m.directory File.join('app/views', file_path_with_mailer)
       m.directory File.join('test/unit', class_path)
-      m.directory File.join('test/fixtures', file_path)
+      m.directory File.join('test/fixtures', file_path_with_mailer)
 
       # Mailer class and unit test.
-      m.template "mailer.rb",    File.join('app/models', class_path, "#{file_name}.rb")
-      m.template "unit_test.rb", File.join('test/unit', class_path, "#{file_name}_test.rb")
+      m.template "mailer.rb",    File.join('app/models', class_path, "#{file_name}_mailer.rb")
+      m.template "unit_test.rb", File.join('test/unit', class_path, "#{file_name}_mailer_test.rb")
 
       # View template and fixture for each action.
       actions.each do |action|
-        relative_path = File.join(file_path, action)
+        relative_path = File.join(file_path_with_mailer, action)
         view_path     = File.join('app/views', "#{relative_path}.erb")
         fixture_path  = File.join('test/fixtures', relative_path)
 
diff --git a/railties/lib/rails_generator/generators/components/mailer/templates/mailer.rb b/railties/lib/rails_generator/generators/components/mailer/templates/mailer.rb
index ce15ae9..9d1eab1 100644
--- a/railties/lib/rails_generator/generators/components/mailer/templates/mailer.rb
+++ b/railties/lib/rails_generator/generators/components/mailer/templates/mailer.rb
@@ -1,4 +1,4 @@
-class <%= class_name %> < ActionMailer::Base
+class <%= class_name %>Mailer < ActionMailer::Base
   
 <% for action in actions -%>
 
-- 
1.5.5

